ansible/vmware_common.py

```diff
@@ -0,0 +1,76 @@
+import os
+import time
+from collections import namedtuple
+from pyVmomi import vim
+
+
+ConnectionData = namedtuple('ConnectionData', ['content', 'datacenter', 'cluster'])
+
+
+def get_obj(content, vim_type, name, folder=None):
+    if not isinstance(vim_type, list):
+        vim_type = [vim_type]
+    if not folder:
+        folder = content.rootFolder
+    container = content.viewManager.CreateContainerView(folder, vim_type, True)
+    return next((obj for obj in container.view if obj.name == name), None)
+
+
+def get_connection_data(service_instance, datacenter_name, cluster_name):
+    content = service_instance.RetrieveContent()
+
+    datacenter = get_obj(content, [vim.Datacenter], datacenter_name)
+    if not datacenter:
+        raise Exception("Couldn't find the Datacenter with the provided name "
+                        "'{}'".format(datacenter_name))
+
+    cluster = get_obj(content, [vim.ClusterComputeResource], cluster_name,
+                      datacenter.hostFolder)
+    if not cluster:
+        raise Exception("Couldn't find the Cluster with the provided name "
+                        "'{}'".format(cluster_name))
+
+    return ConnectionData(content=content, datacenter=datacenter, cluster=cluster)
+
+
+def find_vm(conn, vm_name):
+    vm = get_obj(conn.content, [vim.VirtualMachine], vm_name, conn.datacenter.vmFolder)
+    if not vm:
+        raise Exception("Couldn't find the VirtualMachine with the provided name "
+                        "'{}'".format(vm_name))
+    return vm
+
+
+def find_vm_folder(conn, folder_path):
```

Comment:
Are we actually finding the folder or getting it? I know we call `FindByInventoryPath` below but it looks like `folder_path` is known? If so, maybe renaming this function to `get_vm_folder` would be better.